cmd/secrets-config/README.md

```diff
@@ -0,0 +1,179 @@
+
+This command is used to configure the TLS certificate for external connections, create authentication tokens for inbound proxy access, and other related utility functions.
+
+Proxy configuration commands (listed below) require access to the secret store master key in order to generate temporary secret store access credentials.
+
+# OPTIONS
+
+  * **--confdir** _/path/to/directory/with/configuration.toml_ (optional)
+
+    Points to directory containing a configuration.toml file.
+
+# SUBCOMMANDS
+
+  * **tls**
+
+    Configure inbound TLS certificate. This command will provision the TLS secrets into the secret store and re-deploy them to Kong. Requires additional arguments:
+
+    * **--incert** _/path/to/certchain_ (required)
+
+      Path to TLS leaf certificate (PEM-encoded x.509) (the file extension is arbitrary).
+      If intermediate certificates are required to chain to a certificate authority,
+      these should also be included.
+      The root certificate authority should not be included.
+
+    * **--inkey** _/path/to/private\_key_ (required)
+
+      Path to TLS private key (PEM-encoded).
+
+  * **adduser**
+
+    Create an API gateway user using specified token type. Requires additional arguments:
+
+    * **--token-type** jwt | oauth2 (required)
+    
+      Create user using either the JWT or OAuth2 authentication plugin.
+      This value must match the configured authentication plugin
+      (`KongAuth.Name` in security-proxy-setup's `configuration.toml`).
+
+    * **--user** _username_ (required)
+
+      Username of the user to add.
+
+    * **--group** _group_ (optional)
+
+      Group to which the user belongs, defaults to &quot;admin&quot;.
+      This should be the group associated with the route ACL
+      (`KongAuth.WhiteList` in security-proxy-setup's `configuration.toml`).
+      (Note that secrets-config shares the same configuration as security-proxy-setup
+      as they both configure the EdgeX API gateway.)
+
+
+    The following options are used when token-type == "jwt":
+
+    * **--algorithm** RS256 | ES256 (required for JWT method)
+
+      Algorithm used for signing the JWT.
+      (See [RFC 7518](https://tools.ietf.org/html/rfc7518#section-3.1) for a list of signing algorithms.)
+
+    * **--public\_key** _/path/to/public\_key_ (required for JWT tokens)
+
+      Public key (in PEM format) used to validate the JWT.
+      (Not an x.509 certificate.)
+      This key is assumed to have been pre-created using some external mechanism such as a TPM, HSM, openssl, or other method.
+
+    * **--id** _key_ (optional)
+
+      Optional user-specified &quot;key&quot; used for linkage with an incoming JWT via Kong&#39;s config.key\_claim\_name setting (defaults to &quot;iss&quot; field).
+      See
+      [Kong documentation for JWT plugin](https://docs.konghq.com/hub/kong-inc/jwt/#craft-a-jwt-with-publicprivate-keys-rs256-or-es256)
+      for an example of how this parameter is used.
+
```

Comment:
The OAuth token type section below says that the command will output a JSON struct containing the client_id and client_secret. When I specify jwt as the token type (following your test instructions), the command outputs a simple hex string with no label or description. I think this is the auto-generated key used for when someone doesn't specify `--id`? If so can you mention this here?

Reply:
Documentation added to this effect.